### PR TITLE
Add org.eu.wabarc.wayback

### PR DESCRIPTION
**Pull request by waybackarchiver**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

Related conversations https://github.com/flathub/flathub/pull/3641

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: org.eu.wabarc.wayback.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  <releases>
+    <release date="2022-11-02" version="0.18.1">
+      <description>
+        <p>
+          Please See Official Changelog at github.com/wabarc/wayback/blob/main/CHANGELOG.md
+        </p>
+      </description>
+    </release>
+  </releases>
```
Comment (hfiguiere) [inline org.eu.wabarc.wayback.appdata.xml:39]: still missing mandatory screenshots.

File: org.eu.wabarc.wayback.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+  - --share=network
+  - --filesystem=xdg-documents
+modules:
+  - name: wayback
+    buildsystem: simple
+    build-commands:
+      - |
+        set -eu pipefail
+        version="0.18.1"
+        tar -xvf wayback-linux-*-${version}.tar.gz
```
Comment (hfiguiere) [inline org.eu.wabarc.wayback.yml:16]: what happened to building from source?

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.eu.wabarc.wayback.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        rm -f wayback-linux-*-${version}.tar.gz
+        mv wayback-linux-* wayback
+      - install -Dm755 -t /app/bin wayback
+    sources:
+      - type: file
+        url: https://github.com/wabarc/wayback/releases/download/v0.18.1/wayback-linux-amd64-0.18.1.tar.gz
+        sha256: bca11c1ca33cb03f631efa4d0b304852826ace00580963699520d32e52e7add9
+        only-arches:
+          - x86_64
```
Comment (hfiguiere) [inline org.eu.wabarc.wayback.yml:25]: this doesn't do what you think it does.

there still will be (an attempt) to build an aarch64 flatpak.

File: org.eu.wabarc.wayback.yml [hunk 4]
```diff
@@ -23,11 +23,6 @@ modules:
         sha256: bca11c1ca33cb03f631efa4d0b304852826ace00580963699520d32e52e7add9
         only-arches:
           - x86_64
-      - type: file
-        url: https://github.com/wabarc/wayback/releases/download/v0.18.1/wayback-linux-arm64-0.18.1.tar.gz
-        sha256: 462087adc88f920a06b63d7e127bf822502d7e7c734d0c1c46b5390077276fbf
-        only-arches:
-          - aarch
```
Comment (hfiguiere) [inline org.eu.wabarc.wayback.yml]: this is not what i meant by "invalid arch".

The proper arch id for ARM v8 64-bits is `aarch64`.

File: org.eu.wabarc.wayback.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,48 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline org.eu.wabarc.wayback.appdata.xml:1]: I recommend moving appdata, desktop and icon files to your upstream repo and distribute them in the tarball this gets build from.

**Comment by barthalion** [comment]
bot, build org.eu.wabarc.wayback

**Comment by barthalion** [comment]
I noticed it ships a desktop file but it doesn't actually have any UI? You should change the type in your metainfo file to `console-application` and avoid installing the desktop file at all.

File: org.eu.wabarc.wayback.yml [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  - name: wayback
+    buildsystem: simple
+    build-commands:
+      - |
+        set -eu pipefail
+        version="0.18.1"
+        tar -xvf wayback-linux-*-${version}.tar.gz
+        rm -f wayback-linux-*-${version}.tar.gz
+        mv wayback-linux-* wayback
+        mkdir -p /app/share/applications
```
Comment (hfiguiere) [inline org.eu.wabarc.wayback.yml:19]:
```suggestion
```

This is not necessary. `install -D` will do it for you

**Comment by barthalion** [comment]
bot, build org.eu.wabarc.wayback

File: org.eu.wabarc.wayback.yml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  - --filesystem=xdg-documents
+modules:
+  - name: wayback
+    buildsystem: simple
+    build-commands:
+      - |
+        set -eu pipefail
+        tar -xvf wayback-linux-*.tar.gz
+        mv wayback-linux-* wayback
+      - install -Dm755 -t /app/bin wayback
```
Comment (hfiguiere) [inline org.eu.wabarc.wayback.yml:17]: there is an inconsistency in the path here. line 16 `wayback` doesn't exist, but `mv` expect a directory.
line 17 `wayback` is expected to be a file

**Comment by barthalion** [comment]
bot, build org.eu.wabarc.wayback

**Comment by barthalion** [comment]
Please also add a file named `flathub.json` with the following content:

```
{
  "skip-icons-check": true
}
```

**Comment by barthalion** [comment]
bot, build org.eu.wabarc.wayback

**Pull request closed without merging**